vespajlib/src/main/java/com/yahoo/slime/Injector.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.slime;
+
+/**
+ * @author hakonhall
+ */
+public class Injector {
+    /**
+     * Inject a slime sub-structure described by an Inspector into a slime
+     * structure where the insertion point is described by an
+     * Inserter. This will copy all the values represented by the
+     * Inspector into the position described by the Inserter. Note that
+     * this can be used to either copy data from one Slime structure to
+     * another, or to copy data internally within a single slime
+     * structure. If the Inspector contains the insertion point it will
+     * only be expanded once to avoid infinite recursion.
+     *
+     * @param inspector what to inject
+     * @param inserter where to inject
+     **/
+    public void inject(Inspector inspector, Inserter inserter) {
+        if (inspector.valid()) {
+            injectValue(inserter, inspector, null);
+        }
+    }
+
+    private void injectValue(Inserter inserter, Inspector inspector, Inspector guard) {
```

Comment:
I suggest using the Visitor interface here instead of switch, to be able to differentiate between utf8 and utf16 strings to avoid unneeded conversion.

Reply:
Done